### PR TITLE
Badwords

### PR DESCRIPTION
**Pull request by JulianVennen**

### CODE REVIEW

**Review by matthi4s — CHANGES_REQUESTED** [review]
I would add a "default" bad word response, which uses a predefined string instead of typing the same text over and over again.

File: src/BadWord.js [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+/**
+ * guild wide bad words
+ * @type {module:"discord.js".Collection}
+ */
+const guilds = new Discord.Collection();
+
+/**
+ * Class representing a bad word
+ */
+class BadWord {
```
Comment (matthi4s) [inline src/BadWord.js:30]: It probably makes more sense to create a parent class for `AutoResponse` and `BadWord` to share similar functions like `matches()`.

**Review by matthi4s — CHANGES_REQUESTED** [review]

File: src/BadWord.js (hunk 1 above)
Comment (JulianVennen) [reply src/BadWord.js]: matches() and triggerTypes are the only matching properties I found

**Review by matthi4s — CHANGES_REQUESTED** [review]
I think you can move more stuff e.g. caching + db stuff to the `ChatTriggeredFeature` class by moving the table name and column mapping to different functions.

File: src/features/message/badwordmod.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    if (word.matches(message)) {
+      const reason = `bad word (ID:${word.id})`;
+      await util.delete(message, { reason: reason } );
+      if (word.response !== 'disabled') {
+        const response = await message.reply(word.response === 'default' ? BadWord.defaultResponse : word.response);
+        await util.delete(response, { timeout: 3000 });
+      }
+      await util.logMessageDeletion(message, reason);
+      if (word.punishment.action !== 'none') {
+        await strike.executePunishment(word.punishment, message.guild, message.author, options.bot, options.database, reason);
```
Comment (matthi4s) [inline src/features/message/badwordmod.js:19]: The `reason` for the punishment is visible to the user and should be a bit longer and there is no need to include the bad word id.

File: src/features/message/badwordmod.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const util = require('../../util');
+const strike = require('../../commands/strike');
+
+exports.event = async (options, message) => {
+  if (!message.guild || message.author.bot || await util.isMod(message.member) || message.member.hasPermission("MANAGE_MESSAGES")) return;
+
+  const words = await BadWord.get(message.channel.id, message.guild.id);
+  for (let [,word] of words) {
+    if (word.matches(message)) {
+      const reason = `bad word (ID:${word.id})`;
```
Comment (matthi4s) [inline src/features/message/badwordmod.js:11]:
```suggestion
      const reason = `Bad word (ID:${word.id})`;
```

**Review by matthi4s — CHANGES_REQUESTED** [review]

**Review by matthi4s — DISMISSED** [review]

**Review by KurtThiemann — CHANGES_REQUESTED** [review]
When using variables for table/column names they should be surrounded by ` and properly escaped.
The nodejs MySQL module provides a function to do this: https://www.npmjs.com/package/mysql#escaping-query-identifiers.

**Review by matthi4s — APPROVED** [review]

**Review by KurtThiemann — APPROVED** [review]

**Pull request merged**